pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,483 @@
+        type
+        """
+        return ArrowStringArray
+
+    def __hash__(self) -> int:
+        return hash("ArrowStringDtype")
+
+    def __repr__(self) -> str:
+        return "ArrowStringDtype"
+
+    def __from_arrow__(
+        self, array: Union["pa.Array", "pa.ChunkedArray"]
+    ) -> "ArrowStringArray":
+        """
+        Construct StringArray from pyarrow Array/ChunkedArray.
+        """
+        return ArrowStringArray(array)
+
+    def __eq__(self, other) -> bool:
+        """Check whether 'other' is equal to self.
+
+        By default, 'other' is considered equal if
+        * it's a string matching 'self.name'.
+        * it's an instance of this type.
+
+        Parameters
+        ----------
+        other : Any
+
+        Returns
+        -------
+        bool
+        """
+        if isinstance(other, ArrowStringDtype):
+            return True
+        elif isinstance(other, str) and other == "arrow_string":
+            return True
+        else:
+            return False
+
+
+class ArrowStringArray(ExtensionArray):
+    """
+    Extension array for string data in a ``pyarrow.ChunkedArray``.
+
+    .. versionadded:: 1.2.0
+
+    .. warning::
+
+       ArrowStringArray is considered experimental. The implementation and
+       parts of the API may change without warning.
+
+    Parameters
+    ----------
+    values : pyarrow.Array or pyarrow.ChunkedArray
+        The array of data.
+
+    Attributes
+    ----------
+    None
+
+    Methods
+    -------
+    None
+
+    See Also
+    --------
+    array
+        The recommended function for creating a ArrowStringArray.
+    Series.str
+        The string methods are available on Series backed by
+        a ArrowStringArray.
+
+    Notes
+    -----
+    ArrowStringArray returns a BooleanArray for comparison methods.
+
+    Examples
+    --------
+    >>> pd.array(['This is', 'some text', None, 'data.'], dtype="arrow_string")
+    <ArrowStringArray>
+    ['This is', 'some text', <NA>, 'data.']
+    Length: 4, dtype: arrow_string
+    """
+
+    def __init__(self, values):
+        if isinstance(values, pa.Array):
+            self.data = pa.chunked_array([values])
+        elif isinstance(values, pa.ChunkedArray):
+            self.data = values
+        else:
+            raise ValueError(f"Unsupported type '{type(values)}' for ArrowStringArray")
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        # TODO(ARROW-9407): Accept pd.NA in Arrow
+        scalars_corrected = [None if pd.isna(x) else x for x in scalars]
+        return cls(pa.array(scalars_corrected, type=pa.string()))
+
+    @property
+    def dtype(self) -> ArrowStringDtype:
+        """
+        An instance of 'ArrowStringDtype'.
+        """
+        return ArrowStringDtype()
+
+    def __array__(self, *args, **kwargs) -> "np.ndarray":
+        """Correctly construct numpy arrays when passed to `np.asarray()`."""
+        return self.data.__array__(*args, **kwargs)
```

Comment:
You can also add a `to_numpy()` method (which we need anyway), and use that here instead

Reply:
There are no to_numpy tests in the base extension array tests (except test_to_numpy that does not test with any arguments) or in pandas/tests/arrays/string_/test_string.py

to test the correct output with `dtype`, `copy` and `na_value`

can either add tests to pandas/tests/arrays/string_/test_string.py, parameterise to_numpy tests elsewhere in the test suite and/or do as a precursor. Not yet decided best approach.

> which we need anyway

or as not in scope for the base extension array tests, leave as a followon.

Reply:
I would personally focus on getting the base extension tests passing in this PR, so I am fine leaving full implementation + tests covering all keywords to a follow-up PR, and do only a basic one here.

If you add it here, I would maybe add a new file as `tests/arrays/string_/test_to_numpy.py` (that's similar as what I did for the FloatingArray)

Reply:
> I would personally focus on getting the base extension tests passing in this PR

agreed. as a precursor, have locked down pandas/tests/arrays/string_/test_string.py with xfails as many of the failing tests there are out of scope.

Reply:
> You can also add a `to_numpy()` method (which we need anyway), and use that here instead

OK have done this, but does seem an odd thing to do. The base class defined to_numpy using \_\_array__ (via np.asarray)

Reply:
> The base class defined to_numpy using __array__ (via np.asarray)

That's because it is a "dummy" fallback implementation. The base class also doesn't define `__array__`, so in that case it would fall back to using iter/getitem I think. Here we can be smarter about the exact conversions to numpy.

Reply:
> That's because it is a "dummy" fallback implementation.

I probably wasn't clear about the lack of tests. The fallback option supported `copy` and `na_value`. So have ignored these keywords here, unless also adding tests for expected behaviour.

Reply:
some tests now added. will mark this as resolved and we can move the discussion to https://github.com/pandas-dev/pandas/pull/35259#discussion_r522411045